website/static/js/apiApplication.js

```diff
@@ -0,0 +1,434 @@
+    },
+    deleteApplication: function (appData) {
+        bootbox.confirm({
+            title: 'Deactivate application?',
+            message: language.apiOauth2Application.deactivateConfirm,
+            callback: function (confirmed) {
+                if (confirmed) {
+                    var request = this.client.deleteOne(appData);
+                    request.done(function () {
+                            this.appData.destroy(appData);
+                            var appName = $osf.htmlEscape(appData.name());
+                            $osf.growl('Deletion', '"' + appName + '" has been deactivated', 'success');
+                    }.bind(this));
+                    request.fail(function () {
+                            $osf.growl('Error',
+                                       language.apiOauth2Application.deactivateError,
+                                       'danger');
+                    }.bind(this));
+                }
+            }.bind(this),
+            buttons:{
+                confirm:{
+                    label:'Deactivate',
+                    className:'btn-danger'
+                }
+            }
+        });
+    }
+});
+
+
+/*
+  ViewModel for Detail views (create and update pages- related though distinct behaviors in a single ViewModel)
+    Expects a urls object containing webListUrl, apiListUrl, and apiDetailUrl values. If apiDetailUrl is blank, it
+    behaves like a create view.
+ */
+var ApplicationDetailViewModel = oop.defclass({
+    constructor: function (urls) {
+        var placeholder = new ApplicationData();
+        this.appData = ko.observable(placeholder);
+
+        // Track whether data has changed, and whether user is allowed to leave page anyway
+        this.originalValues = ko.observable(placeholder.serialize());
+        this.dirty = ko.computed(function(){
+            return JSON.stringify(this.originalValues()) !== JSON.stringify(this.appData().serialize());
+        }.bind(this));
+        this.allowExit = ko.observable(false);
+
+        // Set up data access client
+        this.webListUrl = urls.webListUrl;
+        this.client = new ApplicationDataClient(urls.apiListUrl);
+
+        // Control success/failure messages above submit buttons
+        this.message = ko.observable();
+        this.messageClass = ko.observable();
+
+        // Toggle hiding client secret (in detail view)
+        this.showSecret = ko.observable(false);
+        // Toggle display of validation messages
+        this.showMessages = ko.observable(false);
+
+        // // If no detail url provided, render view as though it was a creation form. Otherwise, treat as READ/UPDATE.
+        this.apiDetailUrl = ko.observable(urls.apiDetailUrl);
+        this.isCreateView = ko.computed(function () {
+            return !this.apiDetailUrl();
+        }.bind(this));
+
+        // Prevent user from leaving page if there are unsaved changes
+        $(window).on('beforeunload', function () {
```

Comment:
I don't believe this is necessary. These `beforeunload` checks make sense for forms that *update* existing data because it might not be clear to a user when changes are saved (e.g. some sites autosave on blur, some don't).

Since this for is for *creating* an application, the check is unnecessary; it is clear that the changes will only take effect when you click "Create".